lib/graphql/schema/member/has_fields.rb

```diff
@@ -38,10 +38,24 @@ def get_field(field_name)
           end
         end
 
+        # A list of field names that we should advise users to pick a different
+        # resolve method name.
+        #
+        # @api private
+        CONFLICT_FIELD_NAMES = Set.new([
+          # GraphQL-Ruby conflicts
+          :context, :object,
+          # Ruby built-ins conflicts
+          :method, :class
+        ])
+
         # Register this field with the class, overriding a previous one if needed.
         # @param field_defn [GraphQL::Schema::Field]
         # @return [void]
         def add_field(field_defn)
+          if CONFLICT_FIELD_NAMES.include?(field_defn.method_sym)
+            warn "#{self.graphql_name}'s `field :#{field_defn.name}` conflicts with a built-in method, use `method:` to pick a different resolver method for this field (for example, `method: :resolve_#{field_defn.method_sym}` and `def resolve_#{field_defn.method_sym}`)"
```

Comment:
This should recommend using `resolver_method:` instead of `method:` shouldn't it?

Reply:
yes, thanks 😊